### PR TITLE
Add /getBlob/ endpoint

### PR DESCRIPTION
**Pull request by rafael-atticlabs**

### CODE REVIEW

File: go/datas/remote_database_handlers.go [hunk 1]
```diff
@@ -53,6 +53,11 @@ var (
 	// expects/honors, payload format, and responses.
 	HandleGetRefs = versionCheck(handleGetRefs)
 
+	// HandleGetBlob is a custom endpoint whose sole purpose is to directly fetch the *bytes*
+	// contained in a Blob value. It expects a single query param of `h` to be the ref of the Blob.
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]: 80 character width for go comments

File: go/datas/remote_database_handlers.go [hunk 2]
```diff
@@ -278,6 +283,30 @@ func handleGetRefs(w http.ResponseWriter, req *http.Request, ps URLParams, cs ch
 	}
 }
 
+func handleGetBlob(w http.ResponseWriter, req *http.Request, ps URLParams, cs chunks.ChunkStore) {
+	refStr := req.URL.Query().Get("h")
+	if refStr == "" {
+		d.Panic("Expected h param")
+	}
+
+	fmt.Println("Getting", refStr)
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]: Remove?

File: go/datas/remote_database_handlers.go [hunk 3]
```diff
@@ -278,6 +283,30 @@ func handleGetRefs(w http.ResponseWriter, req *http.Request, ps URLParams, cs ch
 	}
 }
 
+func handleGetBlob(w http.ResponseWriter, req *http.Request, ps URLParams, cs chunks.ChunkStore) {
+	refStr := req.URL.Query().Get("h")
+	if refStr == "" {
+		d.Panic("Expected h param")
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]:
```go
http.Error(w, errors.New("Expected h param"), http.StatusBadRequest)
return
```

File: go/datas/remote_database_handlers.go [hunk 4]
```diff
@@ -278,6 +283,30 @@ func handleGetRefs(w http.ResponseWriter, req *http.Request, ps URLParams, cs ch
...
+	}
+
+	fmt.Println("Getting", refStr)
+	h := hash.Parse(refStr)
+	if (h == hash.Hash{}) {
+		d.Panic("h failed to parse")
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]: BadRequest

File: go/datas/remote_database_handlers.go [hunk 5]
```diff
@@ -278,6 +283,30 @@ func handleGetRefs(w http.ResponseWriter, req *http.Request, ps URLParams, cs ch
...
+	}
+
+	vs := types.NewValueStore(types.NewBatchStoreAdaptor(cs))
+	v := vs.ReadValue(h)
+	b, ok := v.(types.Blob)
+	if !ok {
+		d.Panic("h is not a Blob")
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]: BadRequest

File: go/datas/remote_database_handlers_test.go [hunk 6]
```diff
@@ -245,6 +245,32 @@ func TestHandleGetRefs(t *testing.T) {
...
+		w,
+		newRequest("GET", "", fmt.Sprintf("/getBlob/?h=%s", r.TargetHash().String()), strings.NewReader(""), http.Header{}),
+		params{},
+		cs,
+	)
+
+	if assert.Equal(http.StatusOK, w.Code, "Handler error:\n%s", string(w.Body.Bytes())) {
+		out, _ := ioutil.ReadAll(w.Body)
+		assert.Equal(string(out), blobContents)
+	}
```
Comment (kalman) [inline go/datas/remote_database_handlers_test.go]: would be good to test those error cases (at least missing h and malformed h)

File: go/datas/remote_database_handlers.go (hunk 1 above)
Comment (rafael-atticlabs) [inline go/datas/remote_database_handlers.go]: done

File: go/datas/remote_database_handlers.go (hunk 3 above)
Comment (rafael-atticlabs) [inline go/datas/remote_database_handlers.go]: keeping d.Panic to be consistent

File: go/datas/remote_database_handlers.go (hunk 2 above)
Comment (rafael-atticlabs) [inline go/datas/remote_database_handlers.go]: removed

File: go/datas/remote_database_handlers_test.go (hunk 6 above)
Comment (rafael-atticlabs) [inline go/datas/remote_database_handlers_test.go]: done

File: go/datas/remote_database_handlers.go [hunk 7]
```diff
@@ -44,34 +44,40 @@ var (
 	// ordered sequence of Chunks to be validated and stored on the server.
 	// TODO: Nice comment about what headers it expects/honors, payload
 	// format, and error responses.
-	HandleWriteValue = versionCheck(handleWriteValue)
+	HandleWriteValue = createHanlder(handleWriteValue, true)
```
Comment (kalman) [inline go/datas/remote_database_handlers.go]: Hanlder -> Handler ?

**Pull request merged**